src/redfish_collector/routers/prometheus.py

```
@@ -59,14 +59,28 @@ async def read_all(serverAddress: IPvAnyAddress = Query(None), config: str = Que
+            metrics = generate_latest(registry)
+            return PlainTextResponse(metrics)
+        dataRaw, dataNewSchema, modelSchemaDir = collectorResult
         collectedData = dataReconstructor(dataRaw, dataNewSchema, modelSchemaDir, serverAddress,loglevel)
 
         # dataDir = f'{REDFISH_DATA}NewData/{serverAddress}.json'
         # with open(dataDir, 'r') as file:
         #     collectedData = json.load(file)
 
-        hostName = collectedData['Common'][0]['HostName']
+        if not isinstance(collectedData, dict) or not collectedData.get('Common'):
+            logging.error(f"[{serverAddress}] Reconstructed data has no usable 'Common' section; server data incomplete/unreadable. Reporting target down.")
```

Ruff (0.15.18)

[warning] Logging statement uses f-string

(G004)